src/ansys_sphinx_theme/search/fuse_search.py

```diff
@@ -67,18 +68,60 @@ def iterate_through_docs(self):
                 }
             )
 
+    def construct_title_breadcrumbs(self, section_title: str) -> str:
+        """Generate title breadcrumbs from the document name.
+
+        Parameters
+        ----------
+        section_title : str
+            Section title.
+
+        Returns
+        -------
+        str
+            Title breadcrumbs.
+
+        Notes
+        -----
+        This method is used to generate the title breadcrumbs for the search index.
+        The title breadcrumbs are generated by traversing the document name and
+        getting the title of the document parts.
+        """
+        # removing the last part of the doc_name because it is the file name taken from the nodes
+        docs_parts = self._doc_name.split("/")[:-1]
+
+        title_string = [
+            self.env.titles[doc_part].astext()
+            for doc_part in docs_parts
+            if doc_part in self.env.titles
+        ]
+
+        self.breadcrumbs_title = " > ".join(title_string)
+
+        if self.breadcrumbs_title:
+            if self.section_title == self.doc_title:
+                return f"{self.breadcrumbs_title} > {section_title}"
+            else:
+                return f"{self.breadcrumbs_title} > {self.doc_title} > {section_title}"
+        else:
+            return (
+                section_title
+                if self.section_title == self.doc_title
+                else f"{self.doc_title} > {section_title}"
+            )
```

Comment:
Since there is no code below this `else`, we can remove it and reduce indentation by one level:

```suggestion
        return (
            section_title
            if self.section_title == self.doc_title
            else f"{self.doc_title} > {section_title}"
        )
```